views/js/pciCreator/dev/textReaderInteraction/runtime/textReaderInteraction.amd.js

```diff
@@ -0,0 +1,168 @@
+/*global define,$*/
+define(
+    [
+        'qtiCustomInteractionContext',
+        'OAT/util/event',
+        'textReaderInteraction/runtime/js/renderer'
+    ],
+    function (qtiCustomInteractionContext, event, Renderer) {
+        'use strict';
+        qtiCustomInteractionContext.register({
+            id : -1,
+            getTypeIdentifier : function () {
+                return 'textReaderInteraction';
+            },
+            /**
+             * Render the PCI : 
+             * @param {String} id
+             * @param {Node} dom
+             * @param {Object} config - json
+             */
+            initialize : function (id, dom, config) {
+                var that = this,
+                    pci = this._taoCustomInteraction,
+                    properties = that._taoCustomInteraction.properties,
+                    pageIds = _.pluck(properties.pages, 'id'),
+                    maxPageId = Math.max.apply(null, pageIds);
+
+                this.id = id;
+                this.dom = dom;
+                this.config = config || {};
+                this.$container = $(dom);
+
+                if (!pci.widgetRenderer) {//delivery context
+                    pci.widgetRenderer = new Renderer({
+                        serial : pci.serial,
+                        $container : this.$container
+                    });
+                    pci.widgetRenderer.renderAll(pci.properties);
+                }
```

Comment:
There is a problem here... we want separation of concern : on the one hand there is the runtime code (code for delivery) and on the other hand, we've got the creator widget code (code for the authoring). The runtime code will be packaged into the item runner and delivered to the test taker whereas the authoring code is not supposed to be so. 
Cf. http://forge.taotesting.com/projects/tao/wiki/TAO_PCI_and_PIC#Under-the-hood for sequence diagram.

When implementing the qti creator tool, you can imagine it as two layers, the first layer is represented by this file"textReaderInteraction" and provides the wysiwyg experience and the second interactive layer is added by the creator widget to enable editing its content + properties. Of course the runtime code may (and certainly do) offer some help for the authoring such as dynamic configuration changes or refresh but the main code for authoring should staying the the "creator" directory.
Another argument in favor of this is another tool vendor (or even ourselve later on) may import and reuse your runtime code but implement their authoring widget in a complete different way.

So please reorganize your code to put the pieces of authoring code (+template) in the "creator" part ? And communicate changes by events ?
 TO communicate by events from the authoring to the runtime, you may do something like interaction.triggerPci('yourChangeEventName', [value]); For sample usage, see : https://github.com/oat-sa/extension-tao-itemqti-pci/blob/master/views/js/pciCreator/dev/likertScaleInteraction/creator/widget/states/Question.js
You can also communicate the other way (runtime to authoring). Please let me know if you have any questions.